src/math/integerhistogram1d.cpp

```diff
@@ -0,0 +1,213 @@
+
+// Return maximum value for data
+std::optional<int> IntegerHistogram1D::maximum() const { return maximum_; }
+
+// Bin specified value, returning success
+bool IntegerHistogram1D::bin(int x)
+{
+    if ((minimum_ && x < *minimum_) || (maximum_ && x > *maximum_))
+    {
+        nMissed_++;
+        return false;
+    }
+
+    raw_[x]++;
+    ++nBinned_;
+
+    return true;
+}
+
+// Return number of values binned over all bins
+long int IntegerHistogram1D::nBinned() const { return nBinned_; }
+
+// Accumulate current histogram bins into averages
+void IntegerHistogram1D::accumulate()
+{
+    for (auto &[key, value] : raw_)
+        averages_[key] += (SampledDouble)raw_[key];
+
+    // Update accumulated data
+    updateAccumulatedData();
+}
+
+// Return current data
+Data1D IntegerHistogram1D::data() const
+{
+    Data1D result = accumulatedData_;
+    for (auto &[key, value] : raw_)
+    {
+        result.values()[(key)] = value;
+    }
+    return result;
+}
+
+// Return accumulated (averaged) data
+const Data1D &IntegerHistogram1D::accumulatedData() const { return accumulatedData_; }
+
+/*
+ * Serialisation
+ */
+
+// Read data through specified LineParser
+bool IntegerHistogram1D::deserialise(LineParser &parser)
+{
+    clear();
+    if (parser.getArgsDelim(LineParser::Defaults) != LineParser::Success)
+        return false;
+    std::optional<int> minLim = parser.argb(0) ? std::optional<int>(parser.argi(1)) : std::nullopt;
+    std::optional<int> maxLim = parser.argb(2) ? std::optional<int>(parser.argi(3)) : std::nullopt;
+    initialise(minLim, maxLim);
+
+    if (parser.getArgsDelim(LineParser::Defaults) != LineParser::Success)
+        return false;
+    nBinned_ = parser.argli(0);
+    nMissed_ = parser.argli(1);
+
+    if (parser.getArgsDelim(LineParser::Defaults) != LineParser::Success)
+        return false;
+
+    auto nBins = parser.argli(0);
+
+    if (parser.getArgsDelim(LineParser::Defaults) != LineParser::Success)
+        return false;
+    for (int n = 0; n <= nBins; ++n)
+    {
+        auto bin = parser.argi(0);
+        raw_[bin] = parser.argli(1);
+        averages_[bin].deserialise(parser);
+    }
+    return true;
+}
+
+// Write data through specified LineParser
+bool IntegerHistogram1D::serialise(LineParser &parser) const
+{
+    if (!parser.writeLineF("{} {} {} {}\n", DissolveSys::btoa(minimum_.has_value()), minimum_.value_or(0),
+                           DissolveSys::btoa(maximum_.has_value()), maximum_.value_or(0)))
+        return false;
+    if (!parser.writeLineF("{}  {} \n", nBinned_, nMissed_))
+        return false;
+
+    if (!parser.writeLineF("{} \n", raw_.size()))
+        return false;
+
+    for (auto &[key, value] : raw_)
+    {
+        if (!parser.writeLineF("{} {} \n", key, value))
+            return false;
+    }
+
+    for (auto &[key, value] : averages_)
+    {
+        if (!value.serialise(parser))
+            return false;
+    }
```

Comment:
Need to write the bin counts and averages as pairs, per bin:
```suggestion
    {
        if (!parser.writeLineF("{} {} \n", key, value))
            return false;
        if (!averages_[key].serialise(parser))
            return false;
    }
```